app/assignment/directives/instructorAssignmentTable.js

```diff
@@ -164,21 +174,36 @@ let instructorAssignmentTable = function ($rootScope, AssignmentActionCreators,
 
 				var coursesHtml = "";
 
-				if (scope.showStaffTable == true) {
-					var instructorTypeHeader = '<div class="instructor-type-header">Instructor TBD</div>';
+				if (scope.showInstructorUndeterminedTable == true) {
+					var Unassigned = {};
+					Unassigned.termCodes = {};
+
+					scope.view.state.sectionGroups.ids.forEach(function(sectionGroupId) {
+						var sectionGroup = scope.view.state.sectionGroups.list[sectionGroupId];
+						if (!sectionGroup.isAssigned) {
+							// Scaffold assignments array if this is the first in the termCode
+							if (!Unassigned.termCodes[sectionGroup.termCode]) {
+								Unassigned.termCodes[sectionGroup.termCode] = [];
+							}
+
+							Unassigned.termCodes[sectionGroup.termCode].push(sectionGroup.id);
+						}
+					});
+
+					var theStaffLength = _.keys(scope.view.state.theStaff.termCodes).length;
+					var unassignedLength = _.keys(Unassigned.termCodes).length;
+
+					if (!theStaffLength > 0 && !unassignedLength > 0) {
```

Comment:
Re-write this to be cleaner. Do you really mean to negate the truthiness of theStaffLength and unassignedLength, then compare that truthiness to zero? See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence